### PR TITLE
UI for applications

### LINKED ISSUE #651: Create applications page
### Issue Description

We need to create a page for super user where they can see a list of applications and approve and reject the application.

As per the discussion on 18/12/2023, with Ankush, following feature are to be made in future, after the initial version
- Filter with multiple statuses.
- Sorting by newest first, and oldest first

### PR DESCRIPTION
**Pull request by vinayak-trivedi**
Date:  28/12/2023
----

## Issue Ticket Number
closes #651 

## Description

This PR creates another page for applications, which will be used by super user to accept or reject the applications.

**Documentation Updated?**

-   [ ] Yes
-   [x] No

_If your feature adds a new API, then documentation related to the feature like API-Contracts, and Data-models  must be updated._

**Breaking Changes**

-   [ ] Yes
-   [x] No

_If your feature introduces breaking changes or if something is missing, please mention the related issue tickets._

**Development Tested?**

-   [x] Yes
-   [ ] No

_Confirm whether the changes have been tested locally during development._

**Tested in Staging?**

-   [ ] Yes
-   [x] No

_Indicate whether the changes have been tested in the staging environment._

**Under Feature Flag**

-   [ ] Yes
-   [x] No

_Specify if the changes are currently under a feature flag._

**Database Changes**

-   [ ] Yes
-   [x] No

_Indicate whether the changes include modifications to the database._

### Screenshots
image

image

image

### CODE REVIEW

File: applications/index.html [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        <img
+          class="funnel-icon"
+          src="/taskRequests/assets/funnel.svg"
+          alt="funnel icon"
+        />
+      </button>
+      <p class="no_applications_found hidden">No applications Found!</p>
+      <div
+        class="filter-modal hidden"
+        role="dialog"
```
Comment (anishpawaskar) [inline applications/index.html:29]: please fix this spelling mistake -> `dialogue`

File: applications/script.js [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+
+let status = 'all';
+
+function updateUserApplication({ isAccepted }) {
+  const applicationTextarea = document.querySelector('.application-textarea');
+  let status;
+  const payload = {};
+
+  if (isAccepted) status = 'accepted';
+  else status = 'rejectedd';
```
Comment (anishpawaskar) [inline applications/script.js:43]: please remove extra d from rejectedd

File: applications/script.js [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+
+  if (applicationTextarea.value) payload.feedback = applicationTextarea.value;
+
+  updateApplication({
+    applicationId: currentApplicationId,
+    applicationPayload: payload,
+  })
+    .then((res) => {
+      closeApplicationDetails();
+      alert(res.message);
```
Comment (anishpawaskar) [inline applications/script.js:55]: can you toast instead of alert

File: applications/script.js [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+    })
+    .catch((error) => {
+      alert(error.message);
```
Comment (anishpawaskar) [inline applications/script.js:58]: here also

File: applications/script.js [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+      console.log(error);
```
Comment (anishpawaskar) [inline applications/script.js:59]: remove console

File: applications/script.js [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+    ],
+  };
+
+  const title = createElement({
+    type: 'h1',
+    attributes: { class: 'title' },
+    innerText: `${selectedApplication.username}'s Application`,
+  });
+
+  console.log(application);
```
Comment (anishpawaskar) [inline applications/script.js:127]: remove console

File: applications/script.js [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+      application,
+    });
+    applicationContainer.appendChild(applicationCard);
+  });
+}
+
+(async function renderCardsInitial() {
+  changeLoaderVisibility({ hide: false });
+  const isSuperUser = await getIsSuperUser();
+  console.log(isSuperUser, 'usresr');
```
Comment (anishpawaskar) [inline applications/script.js:273]: remove console here as well

File: applications/utils.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+  const res = await fetch(`${BASE_URL}/users/self`, {
+    method: 'GET',
+    credentials: 'include',
+    headers: {
+      'Content-type': 'application/json',
+    },
+  });
+
+  const self_user = await res.json();
+  console.log(self_user, 'self');
```
Comment (anishpawaskar) [inline applications/utils.js:43]: remove console

**Review by anishpawaskar — COMMENTED** [review]
can you please add a video to show it is responsive or not
also add test stats

File: applications/index.html [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+          Apply Filter
+        </button>
+      </div>
+      <div class="backdrop-blur"></div>
+      <div class="application-details hidden">
+        <button class="application-close-button">
+          <img
+            src="./assets/closeButton.svg"
+            class="close-button-icon"
+            alt=""
```
Comment (satyam73) [inline applications/index.html:75]: NIT: alt can be here

File: applications/script.js [hunk 10]
```diff
@@ -0,0 +1,329 @@
...
+let isDataLoading = false;
+const loader = document.querySelector('.loader');
+const filterButton = document.getElementById('filter-button');
+const filterModal = document.querySelector('.filter-modal');
+const backDrop = document.querySelector('.backdrop');
+const backDropBlur = document.querySelector('.backdrop-blur');
+const applicationDetailsModal = document.querySelector('.application-details');
+const mainContainer = document.querySelector('.container');
+const applicationCloseButton = document.querySelector(
+  '.application-close-button',
```
Comment (satyam73) [inline applications/script.js:18]: why comma is here any specific reason?

File: applications/script.js [hunk 11]
```diff
@@ -0,0 +1,329 @@
...
+  payload['status'] = status;
+
+  if (applicationTextarea.value) payload.feedback = applicationTextarea.value;
+
+  updateApplication({
+    applicationId: currentApplicationId,
+    applicationPayload: payload,
+  })
+    .then((res) => {
+      closeApplicationDetails();
```
Comment (satyam73) [inline applications/script.js:55]: NIT: close application can be in finally block

File: applications/utils.js [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+    url = `${BASE_URL}/applications?size=${size}&status=${applicationStatus}`;
+  }
+
+  const res = await fetch(url, {
+    method: 'GET',
+    credentials: 'include',
+    headers: {
+      'Content-type': 'application/json',
+    },
+  });
```
Comment (satyam73) [inline applications/utils.js:29]: We can use try catch block here

File: __tests__/applications/applications.test.js [hunk 13]
```diff
@@ -0,0 +1,175 @@
+const puppeteer = require('puppeteer');
+
+const {
+  fetchedApplications,
+  acceptedApplications,
+} = require('../../mock-data/applications');
+const { superUserForAudiLogs } = require('../../mock-data/users');
+
+const SITE_URL = 'http://localhost:8000';
```
Comment (satyam73) [inline __tests__/applications/applications.test.js:9]: is site URL will be localhost:8000, please confirm once

File: applications/script.js (hunk 10 above)
Comment (vinayak-trivedi) [reply applications/script.js:18]: no specific reason prettier added it

File: __tests__/applications/applications.test.js (hunk 13 above)
Comment (vinayak-trivedi) [reply __tests__/applications/applications.test.js:9]: for tests it it

**Review by anishpawaskar — APPROVED** [review]

**Review by satyam73 — APPROVED** [review]

**Pull request merged**